contracts/NestedFactory.sol

```diff
@@ -218,21 +217,24 @@ contract NestedFactory is INestedFactory, ReentrancyGuard, OwnableProxyDelegatio
         uint256 buyTokenInitialBalance = _buyToken.balanceOf(address(this));
 
         for (uint256 i = 0; i < tokensLength; i++) {
-            uint256 amount = nestedRecords.getAssetHolding(_nftId, tokens[i]);
-            reserve.withdraw(IERC20(tokens[i]), amount);
+            address token = tokens[i];
+            uint256 amount = nestedRecords.getAssetHolding(_nftId, token);
+            reserve.withdraw(IERC20(token), amount);
 
-            _safeSubmitOrder(tokens[i], address(_buyToken), amount, _nftId, _orders[i]);
-            nestedRecords.freeHolding(_nftId, tokens[i]);
+            _safeSubmitOrder(token, address(_buyToken), amount, _nftId, _orders[i]);
         }
 
         // Amount calculation to send fees and tokens
         uint256 amountBought = _buyToken.balanceOf(address(this)) - buyTokenInitialBalance;
-        uint256 amountFees = amountBought / 100; // 1% Fee
-        amountBought -= amountFees;
 
-        _transferFeeWithRoyalty(amountFees, _buyToken, _nftId);
-        _safeTransferAndUnwrap(_buyToken, amountBought, _msgSender());
+        unchecked {
+            uint256 amountFees = amountBought / 100; // 1% Fee
+            amountBought -= amountFees;
 
+            _transferFeeWithRoyalty(amountFees, _buyToken, _nftId);
+            _safeTransferAndUnwrap(_buyToken, amountBought, _msgSender());
```

Comment:
Is the gas opt worth doing something like this?
From what I read in the documentation, it **should** be safe to englobe the rest of the function in the unchecked scope but it still makes me a bit uncomfortable while reading the code.

Reply:
I added the function calls inside the `unchecked` block because they need to reach the two variables (`amountBought` and `amountFees `), everything declared inside a block is not reachable (`unchecked {...}` or simply `{...}`). 
However, It will not effect the `_transferFeeWithRoyalty` and `_transferFeeWithRoyalty ` that will stay checked, it only affect arithmetic operations : 
![image](https://user-images.githubusercontent.com/22816913/154856808-860cbd11-9f5b-4b35-92ad-9c5759c8232b.png)

[doc](https://docs.soliditylang.org/en/v0.8.0/control-structures.html#checked-or-unchecked-arithmetic)